#3 Bug fix in the option parser and little improvement. Critical bug fix…

andressbarajas opened the pull request:
… of GNU Debugger (GDB) remote interface for MinGW/MSYS.

Pull request number 3 from Sizious dcload-serial Repo. Contains commits:

https://github.com/sizious/dcload-serial/commit/2814d5b1af20c038f71358302af438750116cffa

and

https://github.com/sizious/dcload-serial/commit/1677b5e0f9a11cf598807a622e6ea91bd10b3152

andressbarajas committed:
Bug fix in the option parser and little improvement. Critical bug fix of GNU Debugger (GDB) remote interface for MinGW/MSYS.

andressbarajas committed:
Merge branch 'master' into FixOptionParserGDB

ljsebald commented on host-src/tool/dc-tool.c:
```diff
@@ -284,7 +299,11 @@ char serial_getc()
     retval = serial_read(&tmp, 1);
     if (retval == -1) {
         printf("serial_getc: read error!\n");
-        tmp = 0x00;
+#ifndef __MINGW32__		
+		tmp = (char)NULL;
+#else
+		tmp = 0x00;
+#endif
```
This change shouldn't be needed. I'm pretty sure this is undoing something that was done in one of your earlier PRs.

andressbarajas replied:
Fixed.

ljsebald commented:
There are several instances of changes using tabs instead of spaces for indentation. Please correct.

andressbarajas committed:
Update dc-tool.c

Removed old code

andressbarajas committed:
Update dc-tool.c

Spacing issue

andressbarajas committed:
Update dc-tool.c

Spacing again

andressbarajas committed:
Fixed Spacing

andressbarajas committed:
Fixed Spacing

andressbarajas committed:
Fixed Spacing

andressbarajas commented:
I fixed spacing in dc-tool and syscalls

ljsebald commented on host-src/tool/syscalls.c:
```diff
@@ -486,32 +481,58 @@ void dc_gdbpacket(void)
+        printf( "waiting for gdb client connection...\n" );
+        socket_fd = accept( gdb_server_socket, NULL, NULL );
+#ifdef __MINGW32__
+        if (socket_fd != INVALID_SOCKET)
```
Is this line correct? It just doesn't seem like it is...

andressbarajas replied:
@sizious

andressbarajas replied:
Fixed

ljsebald commented on host-src/tool/syscalls.c:
```diff
@@ -486,32 +481,58 @@ void dc_gdbpacket(void)
 
     if (in_size)
-	write(socket_fd, gdb_buf, in_size);
+#ifdef __MINGW32__
+        send(socket_fd, gdb_buf, in_size, 0);		
+#else		
+        write(socket_fd, gdb_buf, in_size);		
+#endif
```
I suspect that this could use send all around, rather than write...

andressbarajas replied:
@sizious

andressbarajas replied:
Now uses send all around

ljsebald commented on host-src/tool/syscalls.c:
```diff
@@ -486,32 +481,58 @@ void dc_gdbpacket(void)
-	retval = read(socket_fd, gdb_buf, out_size > GDBBUFSIZE ? GDBBUFSIZE : out_size);
-	if (retval == 0)
-	    socket_fd = -1;
+#ifdef __MINGW32__
+        retval = recv(socket_fd, gdb_buf, out_size > GDBBUFSIZE ? GDBBUFSIZE : out_size, 0);
+#else
+        retval = read(socket_fd, gdb_buf, out_size > GDBBUFSIZE ? GDBBUFSIZE : out_size);
+#endif
```
Same goes here. I don't think there's a reason why recv wouldn't work everywhere.

andressbarajas replied:
@sizious

andressbarajas replied:
Now uses recv all around

andressbarajas committed:
Update syscalls.c

Use send/recv all round instead of write/read because there are equivalent with flags set to 0

andressbarajas committed:
Update syscalls.c

Removed unneeded code
